app/services/filter_referees.rb

```diff
@@ -0,0 +1,50 @@
+module Services
+  class FilterReferees
+    attr_accessor :search_query, :filter_by, :relation
+
+    def initialize(params)
+      @search_query = params.delete(:q)
+      @filter_by = params.delete(:filter_by)
+      @relation = Referee.all
+    end
+
+    def filter
+      return relation if filter_by.blank? && search_query.blank?
+
+      sanitize_params
+
+      relation = search_by_name if search_query.present?
+      relation = filter_by_certification if filter_by['certifications'].present?
+      relation = filter_by_national_governing_body if filter_by['national_governing_bodies'].present?
+
+      relation || []
+    end
+
+    private
+
+    def sanitize_params
+      @search_query = JSON.parse(search_query) if search_query.present?
+      @filter_by = JSON.parse(filter_by) if filter_by.present?
+    end
+
+    def search_by_name
+      relation.where('first_name LIKE :query OR last_name LIKE :query', query: "%#{search_query}%")
```

Comment:
I suggest searching by concatenation of first and last name. That way you could also search by full name.